modules/gltf_blend/editor_scene_importer_blend.cpp

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#if TOOLS_ENABLED
+#include "editor_scene_importer_blend.h"
+
+#include "core/core_bind.h"
+#include "core/object/ref_counted.h"
+#include "modules/gltf/editor_scene_importer_gltf.h"
+#include "scene/3d/node_3d.h"
+#include "scene/animation/animation_player.h"
+#include "scene/resources/animation.h"
+
+uint32_t EditorSceneFormatImporterBlend::get_import_flags() const {
+	return ImportFlags::IMPORT_SCENE;
+}
+
+void EditorSceneFormatImporterBlend::get_extensions(List<String> *r_extensions) const {
+	r_extensions->push_back("blend");
+}
+
+Node *EditorSceneFormatImporterBlend::import_scene(const String &p_path,
+		uint32_t p_flags, int p_bake_fps,
+		List<String> *r_missing_deps,
+		Error *r_err) {
+	String addon_path = EditorSettings::get_singleton()->get_setting("filesystem/blend/blender_path");
+	String scene_to_blend_setting = "import bpy, os, sys;"
+									"bpy.ops.wm.open_mainfile(filepath='GODOT_SOURCE');"
+									"(bpy.ops.export_scene.gltf("
+									"filepath='GODOT_SINK',"
+									"export_format='GLTF_SEPARATE',"
+									"export_keep_originals=True,"
+									"export_texture_dir='GODOT_TEXTURE_DIR',"
+									"export_colors=True,"
+									"export_all_influences=False,"
+									"export_extras=True,"
```

Comment:
It would be neat to customize these parameters as scene import options. For example, I often use modifiers (especially `mirror`) so sometimes I'd want `export_apply=True`. However, that prevents exporting shapekeys, so I might want `export_apply=False` on other models (https://docs.blender.org/api/current/bpy.ops.export_scene.html).

I wasn't able to do that when working on a plugin, as it seems like `EditorSceneImporter` can't add custom options like `ResourceImporter` can, so I don't know if it is possible at the moment. If not, would folks be open to a proposal giving `EditorSceneImporter` an equivalent of `EditorImportPlugin.get_import_options`?

Reply:
Or, since `export_apply` is the only one I can think of to customize at the moment, maybe the logic could be:

- If there are modifiers but no shapekeys, set `export_apply=True`
- If there are shapekeys but no modifiers, set `export_apply=False`
- If there are shapekeys and modifiers, raise a warning

That probably requires more python.

Reply:
Reduz said he's working on this after this pr is merged.